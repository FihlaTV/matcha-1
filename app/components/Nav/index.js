import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { MARGINS, COLORS, FONT_SIZES, HEADER_HEIGHT } from 'app/constants/design';
import { injectIntl } from 'react-intl';
import { Actions } from 'react-native-router-flux';

const styles = {
  body: {
    flex: 1,
    justifyContent: 'center',
  },
  nav: {
    backgroundColor: COLORS.BACKGROUND,
    borderColor: COLORS.GREY_LIGHTEST,
    borderTopWidth: 1,
    flexDirection: 'row',
    height: HEADER_HEIGHT,
    justifyContent: 'center',
    paddingHorizontal: MARGINS.SMALL,
  },
  title: {
    color: COLORS.BLUE_DARKER,
    fontSize: FONT_SIZES.BIG,
    textAlign: 'center',
  },
};

type Props = {
  intl: any,
};

const Nav = (props: Props) => (
  <View style={styles.nav}>
    <TouchableOpacity style={styles.body} onPress={() => Actions.replace('feed')}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Nav.home' })}</Text>
    </TouchableOpacity>
    <TouchableOpacity style={styles.body} onPress={() => Actions.replace('discover')}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Nav.discover' })}</Text>
    </TouchableOpacity>
    <TouchableOpacity style={styles.body} onPress={() => Actions.replace('messages')}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Nav.messages' })}</Text>
    </TouchableOpacity>
    <TouchableOpacity style={styles.body} onPress={() => Actions.replace('profile')}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Nav.profile' })}</Text>
    </TouchableOpacity>
  </View>
);
export default injectIntl(Nav);
