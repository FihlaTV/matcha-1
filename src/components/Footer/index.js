import React from 'react';
import { View, Text } from 'react-native';
import { MARGINS, COLORS, FONT_SIZES, HEADER_HEIGHT } from 'app/constants/design';
import { injectIntl } from 'react-intl';

const styles = {
  body: {
    flex: 1,
    justifyContent: 'center',
  },
  footer: {
    backgroundColor: COLORS.BACKGROUND,
    flexDirection: 'row',
    height: HEADER_HEIGHT,
    justifyContent: 'center',
    paddingHorizontal: MARGINS.SMALL,
  },
  title: {
    color: COLORS.BLUE_DARKER,
    fontSize: FONT_SIZES.BIG,
    textAlign: 'center',
  },
};

type Props = {
  intl: any,
};

const Footer = (props: Props) => (
  <View style={styles.footer}>
    <View style={styles.body}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Footer.home' })}</Text>
    </View>
    <View style={styles.body}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Footer.messages' })}</Text>
    </View>
    <View style={styles.body}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Footer.profile' })}</Text>
    </View>
  </View>
);
export default injectIntl(Footer);
export { Footer };

