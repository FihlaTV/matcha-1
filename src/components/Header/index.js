// @flow
import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { MARGINS, COLORS, FONT_SIZES, HEADER_HEIGHT } from 'src/constants/design';
import { injectIntl } from 'react-intl';
import Icon from 'react-native-vector-icons/FontAwesome';
import { Actions } from 'react-native-router-flux';

const styles = {
  body: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  header: {
    backgroundColor: COLORS.BACKGROUND,
    borderBottomWidth: 1,
    borderColor: COLORS.GREY_LIGHTEST,
    flexDirection: 'row',
    height: HEADER_HEIGHT,
    justifyContent: 'center',
    position: 'absolute',
    width: '100%',
    zIndex: 1,
  },
  left: {
    alignItems: 'flex-start',
    flex: 1,
    justifyContent: 'center',
    paddingHorizontal: MARGINS.SMALL,
  },
  right: {
    alignItems: 'flex-end',
    flex: 1,
    justifyContent: 'center',
    paddingHorizontal: MARGINS.SMALL,
  },
  title: {
    color: COLORS.BLUE_DARKER,
    fontSize: FONT_SIZES.BIG,
    textAlign: 'center',
  },
};

type Props = {
  intl: any,
  hasBack?: boolean,
  title?: string,
  actions: Array<Node>,
};

const Header = (props: Props) => (
  <View style={styles.header}>
    <View style={styles.left}>
      { props.hasBack &&
        <TouchableOpacity onPress={() => Actions.pop()}>
          <Icon name="chevron-left" size={20} color={COLORS.BLUE_DARKER} />
        </TouchableOpacity>
      }
    </View>
    <View style={styles.body}>
      <Text style={styles.title}>
        {
          props.title
            ? props.title
            : props.intl.formatMessage({ id: 'App.title' })
        }
      </Text>
    </View>
    <View style={styles.right}>
      {props.actions.map((action) => action)}
    </View>
  </View>
);

Header.defaultProps = {
  hasBack: false,
  title: undefined,
};

export default injectIntl(Header);
