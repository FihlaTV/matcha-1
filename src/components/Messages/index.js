// @flow
import React from 'react';
import { Text, View, StyleSheet, Image } from 'react-native';
import type { MessageType } from 'src/types/Message';
import { MARGINS, COLORS } from 'src/constants/design';
import moment from 'moment';
import { injectIntl, type IntlShape } from 'react-intl';
import LoadingHOC from 'src/components/HOC/LoaderHOC';

const styles = StyleSheet.create({
  leftBubble: {
    alignSelf: 'flex-start',
    flexDirection: 'row',
    marginTop: MARGINS.TINY,
  },
  leftBubbleText: {
    alignSelf: 'flex-start',
    backgroundColor: COLORS.GREY_LIGHTEST,
    borderRadius: 20,
    flexDirection: 'row',
    flexWrap: 'wrap',
    paddingHorizontal: MARGINS.SMALL,
    paddingVertical: MARGINS.TINY,
  },
  messageContainer: {
    alignItems: 'center',
    backgroundColor: COLORS.BACKGROUND,
    padding: MARGINS.SMALL,
  },
  messageProfile: {
    backgroundColor: COLORS.GREY_DARK,
    borderRadius: 20,
    height: 40,
    marginRight: MARGINS.SMALL,
    width: 40,
  },
  messagesContainer: {
    flex: 1,
  },
  photoProfile: {
    backgroundColor: COLORS.GREY_DARK,
    borderRadius: 10,
    height: 20,
    marginRight: MARGINS.SMALL,
    marginTop: MARGINS.TINY,
    width: 20,
  },
  rightBubble: {
    alignSelf: 'flex-end',
    backgroundColor: COLORS.BLUE_LIGHT,
    borderRadius: 20,
    flexDirection: 'row',
    marginTop: MARGINS.TINY,
    paddingHorizontal: MARGINS.SMALL,
    paddingVertical: MARGINS.TINY,
  },
  timeText: {
    color: COLORS.GREY_DARK,
    fontSize: 12,
  },
});

type MessageProps = {
  message: MessageType,
};

const Message = (props: MessageProps) => (
  <View style={styles.messageContainer}>
    <Text style={styles.timeText}>
      {
        moment(Date.now()).diff(moment(props.message.createdAt), 'days') > 5 // eslint-disable-line
          ? moment(props.message.createdAt).format('dddd, MMM Do YYYY, h:mm:ss')
          : moment(props.message.createdAt).fromNow()
      }
    </Text>
    {
      props.message.from.username === 'admin' ?
        <View style={styles.rightBubble}>
          <Text style={{ color: COLORS.WHITE }}>{props.message.message}</Text>
        </View> :
        <View style={styles.leftBubble}>
          {/* eslint-disable-next-line */}
          <Image style={styles.photoProfile} source={{ uri: props.message.from.picture }} defaultSource={require('src/assets/images/no-pic.jpg')} />
          <View style={styles.leftBubbleText}>
            <Text style={{ flexWrap: 'wrap', maxWidth: 250 }}>{props.message.message}</Text>
          </View>
        </View>
    }
  </View>
);

type MessagesProps = {
  intl: IntlShape,
  messages: Array<MessageType>,
};

const Messages = (props: MessagesProps) => (
  <View style={styles.messagesContainer}>
    {props.messages.length > 0 &&
      <View style={styles.messageContainer}>
        <Text style={styles.timeText}>
          {moment(props.messages[0].createdAt).format('dddd, MMM Do YYYY, h:mm:ss')}
        </Text>
        <Text style={{ color: COLORS.GREY_DARK, marginTop: MARGINS.TINY, textAlign: 'center' }}>
          {props.intl.formatMessage({ id: 'Messages.match' })}
        </Text>
      </View>}
    {props.messages.map((message) =>
      (
        !message.match.data[0]
          ? <Message key={message.id} message={message} />
          : null
      ))
    }
  </View>
);

export default injectIntl(LoadingHOC('messages')(Messages));
export { Messages, Message };
