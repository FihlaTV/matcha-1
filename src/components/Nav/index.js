import React from 'react';
import { View, Text } from 'react-native';
import { MARGINS, COLORS, FONT_SIZES, HEADER_HEIGHT } from 'app/constants/design';
import { injectIntl } from 'react-intl';

const styles = {
  body: {
    flex: 1,
    justifyContent: 'center',
  },
  nav: {
    backgroundColor: COLORS.BACKGROUND,
    flexDirection: 'row',
    height: HEADER_HEIGHT,
    justifyContent: 'center',
    paddingHorizontal: MARGINS.SMALL,
  },
  title: {
    color: COLORS.BLUE_DARKER,
    fontSize: FONT_SIZES.BIG,
    textAlign: 'center',
  },
};

type Props = {
  intl: any,
};

const Nav = (props: Props) => (
  <View style={styles.nav}>
    <View style={styles.body}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Nav.home' })}</Text>
    </View>
    <View style={styles.body}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Nav.discover' })}</Text>
    </View>
    <View style={styles.body}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Nav.messages' })}</Text>
    </View>
    <View style={styles.body}>
      <Text style={styles.title}>{props.intl.formatMessage({ id: 'Nav.profile' })}</Text>
    </View>
  </View>
);
export default injectIntl(Nav);
export { Nav };

