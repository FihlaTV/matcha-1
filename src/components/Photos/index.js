import React from 'react';
import { Text, View, StyleSheet, Image } from 'react-native';
import type { PhotoType } from 'app/types/Photo';
import { MARGINS, COLORS } from 'app/constants/design';
import Icon from 'react-native-vector-icons/FontAwesome';
import moment from 'moment';

const styles = StyleSheet.create({
  photoContainer: {
    backgroundColor: COLORS.BACKGROUND,
    borderColor: COLORS.GREY,
    borderWidth: 1,
    margin: MARGINS.SMALL,
    padding: MARGINS.SMALL,
  },
  photoControl: {
    flexDirection: 'row',
    marginBottom: MARGINS.SMALL,
  },
  photoHeader: {
    flexDirection: 'row',
    marginBottom: MARGINS.SMALL,
  },
  photoImage: {
    backgroundColor: COLORS.GREY_DARK,
    height: 325,
    margin: 0,
    marginBottom: MARGINS.SMALL,
    width: 325,
  },
  photoProfile: {
    backgroundColor: COLORS.GREY_DARK,
    borderRadius: 20,
    height: 20,
    marginRight: MARGINS.SMALL,
    width: 20,
  },
  photosContainer: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
  },
});

type PhotoProps = {
  photo: PhotoType,
};

const Photo = (props: PhotoProps) => (
  <View style={styles.photoContainer}>
    <View style={styles.photoHeader}>
      <Image style={styles.photoProfile} source={props.photo.user.picture} />
      <Text>{props.photo.user.username}</Text>
    </View>
    <Image style={styles.photoImage} source={props.photo.url} />
    <View style={styles.photoControl}>
      <Icon style={{ marginRight: MARGINS.SMALL }} name="heart" size={20} color={COLORS.GREY_DARK} />
      <Icon name="comment" size={20} color={COLORS.GREY_DARK} />
    </View>
    <Text>{moment(props.photo.createdAt).fromNow()}</Text>
  </View>
);

type PhotosProps = {
  photos: Array<PhotoType>,
};

const Photos = (props: PhotosProps) => (
  <View style={styles.photosContainer}>
    {
      props.photos.map((photo) => (<Photo key={photo.id} photo={photo} />))
    }
  </View>
);

export default Photos;
export { Photos };
