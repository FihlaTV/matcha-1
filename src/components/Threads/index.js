// @flow
import React from 'react';
import { Text, View, StyleSheet, Image, ScrollView, TouchableOpacity } from 'react-native';
import type { ThreadType } from 'src/types/Thread';
import { MARGINS, COLORS } from 'src/constants/design';
import moment from 'moment';
import LoadingHOC from 'src/components/HOC/LoaderHOC';

const styles = StyleSheet.create({
  threadContainer: {
    alignItems: 'center',
    backgroundColor: COLORS.BACKGROUND,
    borderBottomWidth: 1,
    borderColor: COLORS.GREY_LIGHTEST,
    flexDirection: 'row',
    padding: MARGINS.SMALL,
  },
  threadHeader: {
    flex: 1,
    flexDirection: 'row',
    marginBottom: MARGINS.TINY,
  },
  threadMatchProfile: {
    backgroundColor: COLORS.GREY_DARK,
    borderRadius: 30,
    height: 60,
    margin: MARGINS.SMALL,
    width: 60,
  },
  threadProfile: {
    backgroundColor: COLORS.GREY_DARK,
    borderRadius: 20,
    height: 40,
    marginRight: MARGINS.SMALL,
    width: 40,
  },
  threadsContainer: {
    flex: 1,
  },
  threadsMatchContainer: {
    borderBottomWidth: 1,
    borderColor: COLORS.GREY_LIGHTEST,
    flexDirection: 'row',
  },
});

type ThreadProps = {
  thread: ThreadType,
};

const Thread = (props: ThreadProps) => (
  <View style={styles.threadContainer}>
    {/* eslint-disable-next-line */}
    <Image style={styles.threadProfile} source={{ uri: props.thread.user2.picture }} defaultSource={require('src/assets/images/no-pic.jpg')} />
    <View style={{ flex: 1 }}>
      <View style={styles.threadHeader}>
        <Text>{props.thread.user2.name}</Text>
        <Text style={{ flex: 1, textAlign: 'right' }}>
          {moment(props.thread.createdAt).fromNow()}
        </Text>
      </View>
      <Text>{props.thread.lastMessage.message}</Text>
    </View>
  </View>
);

type ThreadMatchProps = {
  thread: ThreadType,
};

const ThreadMatch = (props: ThreadMatchProps) => (
  <Image style={styles.threadMatchProfile} source={{ uri: props.thread.user2.picture }} defaultSource={require('src/assets/images/no-pic.jpg')} /> //eslint-disable-line
);

type ThreadsProps = {
  threads: Array<ThreadType>,
  onSelectThread: (string) => {}, //eslint-disable-line
};

const Threads = (props: ThreadsProps) => (
  <View style={styles.threadsContainer}>
    <ScrollView horizontal style={styles.threadsMatchContainer}>
      {props.threads.map((thread) =>
        (
          thread.lastMessage.match.data[0] ?
            <TouchableOpacity key={thread.id} onPress={() => props.onSelectThread(thread.id)}>
              <ThreadMatch thread={thread} />
            </TouchableOpacity>
            : null
        ))
      }
    </ScrollView>
    <View style={styles.threadsContainer}>
      {props.threads.map((thread) =>
        (
          !thread.lastMessage.match.data[0] ?
            <TouchableOpacity key={thread.id} onPress={() => props.onSelectThread(thread.id)}>
              <Thread thread={thread} />
            </TouchableOpacity>
            : null
        ))
      }
    </View>
  </View>
);

export default LoadingHOC('threads')(Threads);
export { Threads, Thread, ThreadMatch };
