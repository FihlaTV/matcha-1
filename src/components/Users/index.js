// @flow
import React from 'react';
import { Text, View, StyleSheet, Image } from 'react-native';
import { MARGINS, COLORS } from 'src/constants/design';
import Icon from 'react-native-vector-icons/FontAwesome';
import moment from 'moment';
import geolib from 'geolib';
import LoadingHOC from 'src/components/HOC/LoaderHOC';

const styles = StyleSheet.create({
  userContainer: {
    backgroundColor: COLORS.BACKGROUND,
    borderColor: COLORS.GREY_LIGHTEST,
    borderWidth: 1,
    margin: MARGINS.SMALL,
    padding: MARGINS.SMALL,
  },
  userControl: {
    alignSelf: 'center',
    bottom: 50,
    flexDirection: 'row',
    marginBottom: MARGINS.SMALL,
    position: 'absolute',
  },
  userHeader: {
    flexDirection: 'row',
    marginBottom: MARGINS.SMALL,
  },
  userImage: {
    alignItems: 'center',
    backgroundColor: COLORS.GREY_DARK,
    height: 325,
    justifyContent: 'flex-end',
    margin: 0,
    marginBottom: MARGINS.SMALL,
    width: 325,
  },
  userProfile: {
    backgroundColor: COLORS.GREY_DARK,
    borderRadius: 10,
    height: 20,
    marginRight: MARGINS.SMALL,
    width: 20,
  },
  usersContainer: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
  },
});

const getDistance = (lat1, long1, lat2, long2) => {
  const distance = geolib.getDistance(
    { latitude: lat1, longitude: long1 },
    { latitude: lat2, longitude: long2 },
  );
  /* eslint-disable-next-line */
  return distance < 500 ? `${distance} meters away` : `${distance/1000} kilometers away`;
};

type UserProps = {
  user: UserType,
  currentUser?: UserType,
};

const User = (props: UserProps) => (
  <View style={styles.userContainer}>
    <View style={styles.userHeader}>
      <Text>{props.user.name}</Text>
    </View>
    {/* eslint-disable-next-line */}
    <Image style={styles.userImage} source={{ uri: props.user.picture }} defaultSource={require('src/assets/images/no-pic.jpg')} />
    {
      props.currentUser &&
      <Text>{getDistance(props.user.lat, props.user.long, props.currentUser.lat, props.currentUser.long)}</Text>
    }
  </View>
);

type UsersProps = {
  users: Array<UserType>,
  currentUser: UserType,
};

const Users = (props: UsersProps) => (
  <View style={styles.usersContainer}>
    {
      props.users.map((user) =>
        (user.id !== props.currentUser.id ?
          <View key={user.id}>
            <User user={user} currentUser={props.currentUser} />
            <View style={styles.userControl}>
              <Icon style={{ marginRight: MARGINS.SMALL }} name="heart" size={40} color={COLORS.RED} />
              <Icon name="times-circle" size={40} color={COLORS.RED} />
            </View>
          </View>
          : null
        ))
    }
  </View>
);

User.defaultProps = {
  currentUser: undefined,
};

export default LoadingHOC('users')(Users);
export { Users, User };
