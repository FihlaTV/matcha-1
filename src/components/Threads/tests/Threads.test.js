import React from 'react';
import { Threads } from 'app/components/Threads';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Threads', () => {
  describe('render test', () => {
    it('should render', () => {
      const wrapper = shallow(<Threads threads={[]} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
