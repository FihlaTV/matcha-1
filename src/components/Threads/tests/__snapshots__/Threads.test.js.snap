// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Threads render test should render 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    horizontal={true}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  />
</Component>
`;
